src/notification.c

```diff
@@ -382,6 +363,122 @@ check_notification (GVariant *notification,
   return TRUE;
 }
 
+static gboolean
+validate_icon_more (GVariant *v)
+{
+  g_autoptr(GIcon) icon = g_icon_deserialize (v);
+  GBytes *bytes;
+  g_autofree char *name = NULL;
+  const char *argv[] = { LIBEXECDIR "/xdg-desktop-portal-validate-icon", "name", NULL };
```

Comment:
Do we want to wrap it in `bwrap` just to completely eliminate a ton of possible exploits? I mean we know the system has it might as well use it.